### PR TITLE
v6.0 deprecated features

### PR DESCRIPTION
**Pull request by amyblais**
While waiting for v6.0 communications, I'll start listing v6.0 deprecated features in the release notes (will also add this list in the deprecated features webpage once this PR is reviewed).

I'll take a look if there are other places to share this information while waiting for the official comms.

### CODE REVIEW

**Comment by amyblais** [comment]
@wiersgallak Can we plan to merge this while the other comms are in progress?

File: source/administration/changelog.md [hunk 1]
```diff
@@ -127,6 +127,34 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
 
+### Upcoming Deprecations in Mattermost v6.0
+
+The following deprecations are planned for the Mattermost v6.0 release, which is scheduled for 2021/09/15. This list is subject to change prior to the release.
+
+1. [Legacy Command Line Tools](https://docs.mattermost.com/administration/command-line-tools.html). All commands have been fully replaced by [MMCTL](https://docs.mattermost.com/administration/mmctl-cli-tool.html) and new commands have been added over the last few months, making this tool a full and robust replacement. 
+
+2. [Slack Import via the Web App](https://docs.mattermost.com/administration/migrating.html?highlight=mmetl#migrating-from-slack-using-the-mattermost-web-app). The Slack import tool accessible via the Team Setting menu is being replaced by the [MMETL](https://docs.mattermost.com/administration/migrating.html?highlight=mmetl#migrating-from-slack-using-the-mattermost-mmetl-tool-and-bulk-import) tool that is much more comprehensive for types of data it can assist in uploading. 
+
+3. PostgreSQL versions prior 10.0 - [versions 9.5 EOL in February 2021 and 9.6 will be EOL in November 2021](https://www.postgresql.org/support/versioning/).
```
Comment (wiersgallak) [inline source/administration/changelog.md:138]: This can be removed.  Claudio confirmed this was already implemented.

**Review by wiersgallak — COMMENTED** [review]
Hi Amy - one outstanding comment from me and I am good with publishing. Thanks!

**Review by wiersgallak — APPROVED** [review]

File: source/administration/changelog.md [hunk 2]
```diff
@@ -127,6 +127,32 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+
+1. [Legacy Command Line Tools](https://docs.mattermost.com/administration/command-line-tools.html). All commands have been fully replaced by [MMCTL](https://docs.mattermost.com/administration/mmctl-cli-tool.html) and new commands have been added over the last few months, making this tool a full and robust replacement. 
+
+2. [Slack Import via the Web App](https://docs.mattermost.com/administration/migrating.html?highlight=mmetl#migrating-from-slack-using-the-mattermost-web-app). The Slack import tool accessible via the Team Setting menu is being replaced by the [MMETL](https://docs.mattermost.com/administration/migrating.html?highlight=mmetl#migrating-from-slack-using-the-mattermost-mmetl-tool-and-bulk-import) tool that is much more comprehensive for types of data it can assist in uploading. 
+
+3. MySQL 5.6 which reached [EOL in February 2021](https://www.mysql.com/support/eol-notice.html).
+
+4. Elasticsearch 5 & 6 - [versions 5.x EOL in March of 2019, versions 6.x EOL in November 2020](https://www.elastic.co/support/eol).
```
Comment (cwarnermm) [inline source/administration/changelog.md:140]:
```suggestion
4. Elasticsearch 5 & 6 - [versions 5.x reached EOL in March of 2019, and versions 6.x reached EOL in November 2020](https://www.elastic.co/support/eol).
```

File: source/administration/changelog.md [hunk 3]
```diff
@@ -127,6 +127,32 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+
+5. Windows 7 reached [EOL in January 2020](https://support.microsoft.com/en-us/windows/windows-7-support-ended-on-january-14-2020-b75d4580-2cc7-895a-2c9c-1466d9a53962). We will no longer provide support for desktop app issues on Windows 7.
```
Comment (cwarnermm) [inline source/administration/changelog.md:142]:
```suggestion
5. Windows 7 reached [EOL in January 2020](https://support.microsoft.com/en-us/windows/windows-7-support-ended-on-january-14-2020-b75d4580-2cc7-895a-2c9c-1466d9a53962). We will no longer provide support for the Desktop App issues on Windows 7.
```

File: source/administration/changelog.md [hunk 4]
```diff
@@ -127,6 +127,32 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+
+6. [DisablePostMetaData](https://docs.mattermost.com/administration/config-settings.html#disable-post-metadata) https://docs.mattermost.com/administration/config-settings.html#deprecated-configuration-settings configuration setting.
+
+7. [Experimental Timezone](https://docs.mattermost.com/administration/config-settings.html#timezone) configuration setting.
+
+8. [All configuration that has been previously marked “Deprecated”](https://docs.mattermost.com/administration/config-settings.html#deprecated-configuration-settings) configuration setting.
```
Comment (cwarnermm) [inline source/administration/changelog.md:148]:
```suggestion
8. [All configuration settings previously marked as “Deprecated”](https://docs.mattermost.com/administration/config-settings.html#deprecated-configuration-settings).
```

File: source/administration/changelog.md [hunk 5]
```diff
@@ -127,6 +127,32 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+
+9. Remove deprecated ``model.CommandArgs.Session``.
+
+10. Restrict ``UserWillLogIn``, ``FileWillBeUploaded``, and ``MessageWillBe[Posted|Updated]`` hooks to enterprise licenced instances only.
```
Comment (cwarnermm) [inline source/administration/changelog.md:152]:
```suggestion
10. Restrict ``UserWillLogIn``, ``FileWillBeUploaded``, and ``MessageWillBe[Posted|Updated]`` hooks to Mattermost Enterprise instances only.
```

**Review by cwarnermm — COMMENTED** [review]
Suggestions provided inline

**Review by cwarnermm — APPROVED** [review]

File: source/administration/changelog.md [hunk 6]
```diff
@@ -127,7 +127,34 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+
+6. [DisablePostMetaData](https://docs.mattermost.com/administration/config-settings.html#disable-post-metadata) configuration setting.
+
+7. [Experimental Timezone](https://docs.mattermost.com/administration/config-settings.html#timezone) configuration setting.
+
+8. [All configuration settings previously marked as “Deprecated”](https://docs.mattermost.com/administration/config-settings.html#deprecated-configuration-settings).
+
+9. Remove deprecated ``model.CommandArgs.Session``.
+
+10. Restrict ``UserWillLogIn``, ``FileWillBeUploaded``, and ``MessageWillBe[Posted|Updated]`` hooks to Mattermost Enterprise instances only.
```
Comment (hanzei) [inline source/administration/changelog.md:152]: These two items are still under discussion and it's not clear (yet) if they will apply for 6.0.

**Review by hanzei — CHANGES_REQUESTED** [review]

File: source/administration/changelog.md (hunk 6 above)
Comment (amyblais) [reply source/administration/changelog.md:152]: @hanzei @aaronrothschild When is the decision made? Breaking changes / deprecated features are normally communicated to customers 3 months ahead of the release and I'm planning to merge this PR by this week.

Comment (amyblais) [reply source/administration/changelog.md]: I've removed them for now, but would like clarity on my question.

File: source/administration/changelog.md [hunk 7]
```diff
@@ -127,7 +127,30 @@ Multiple setting options were added to ``config.json``. Below is a list of the a
...
+3. MySQL versions below to 5.7.7. Minimum support will now be for 5.7.8. This version introduced a native JSON data type that lets us improve performance and scalability of several database fields (most notably Users and Posts props), Additionally, 5.6 (our current minimum version) reached [EOL in February 2021](https://www.mysql.com/support/eol-notice.html).
+
+4. Elasticsearch 5 & 6 - [versions 5.x reached EOL in March of 2019, and versions 6.x reached EOL in November 2020](https://www.elastic.co/support/eol).
+
+5. Windows 7 reached [EOL in January 2020](https://support.microsoft.com/en-us/windows/windows-7-support-ended-on-january-14-2020-b75d4580-2cc7-895a-2c9c-1466d9a53962). We will no longer provide support for the Desktop App issues on Windows 7.
+
+6. [DisablePostMetaData](https://docs.mattermost.com/administration/config-settings.html#disable-post-metadata) configuration setting.
+
+7. [Experimental Timezone](https://docs.mattermost.com/administration/config-settings.html#timezone) configuration setting.
+
```
Comment (esethna) [inline source/administration/changelog.md:147]: I think we're missing a lot of configs here: https://focalboard-community.octo.mattermost.com/workspace/suciie6fxpbz9rryiwuxqa8yqh?id=0bdf3bf4-dc53-4274-88a6-121003f431e4&v=8763e825-c543-4ec5-809a-73fa705dc2a8&c=247af414-fcf9-4c6b-857e-564fe49c16d3

**Review by esethna — CHANGES_REQUESTED** [review]

File: source/administration/changelog.md (hunk 7 above)
Comment (amyblais) [reply source/administration/changelog.md:147]: @esethna The list comes from the official communications from PMs https://docs.google.com/document/d/13XseTWS2Bp7hsmiX-6AqMkiFMRwQOqMduNPpUizjdR8/edit. @wiersgallak can confirm.

Comment (esethna) [reply source/administration/changelog.md:147]: The "legacy sidebar" items listed in that doc are all these: 
image

**Review by hanzei — APPROVED** [review]

**Comment by amyblais** [comment]
@esethna Can you help approve so that I can merge this?

**Review by esethna — APPROVED** [review]

**Pull request merged**